lib/core/test/unit/Cardano/Pool/RegistrySpec.hs

```diff
@@ -0,0 +1,277 @@
+            res `shouldBe` Right (map Just presentMetas)
+
+        it "Handles a missing pool" $ \port -> do
+            tr <- setupLogging
+            res <- getStakePoolMetadata tr (testUrl port) (absentPid:presentPids)
+            res `shouldBe` Right (Nothing:map Just presentMetas)
+            
+        it "Fails with an unavailable HTTP server" $ \_port -> do
+            tr <- setupLogging
+            let badUrl = "http://localhost:99/master.zip"
+            res <- getStakePoolMetadata tr badUrl presentPids
+            case res of
+                Left (FetchErrorDownload msg) ->
+                    head (words msg) `shouldBe` "HttpExceptionRequest"
+                _ -> error "expected fetch error but got none"
+                
+    describe "Cardano.Pool.Registry.getStakePoolMetadata" $
+        it "Arbitrary zip files" $
+            property prop_getStakePoolMetadata
+   
+{-------------------------------------------------------------------------------
+                                 Test fixtures
+-------------------------------------------------------------------------------}
+
+-- | Run a HTTP file server on any free port
+testServer :: FilePath -> (Port -> IO a) -> IO a
+testServer root = withApplication (pure app)
+    where app = staticApp $ defaultWebAppSettings root
+
+dataDir :: FilePath
+dataDir = $( TH.LitE . TH.StringL <$>
+    makeRelativeToProject "test/data/stake-pool-registry" )
+
+-- | Make a file server URL for the test data file.
+-- This file was downloaded from <https://github.com/input-output-hk/testnet-stake-pool-registry/archive/master.zip>
+testUrl :: Port -> String
+testUrl p = "http://localhost:" <> show p <> "/testnet-stake-pool-registry-master.zip"
+
+presentPids :: [PoolId]
+presentPids = map PoolId
+    [ "pk1afhcpw2tg7nr2m3wr4x8jaa4dv7d09gnv27kwfxpjyvukwxs8qdqwg85xp"
+    , "pk1z4vh8gva25w07x8574uujuveu8gz43fu6qfln3t4prcavrvcphjsk0pdqs"
+    ]
+
+presentMetas :: [StakePoolMetadata]
+presentMetas =
+    [ StakePoolMetadata (unsafeFromText "FST") "https://12345"
+        "ed25519_pk15vz9yc5c3upgze8tg5kd7kkzxqgqfxk5a3kudp22hdg0l2za00sq2ufkk7"
+    , StakePoolMetadata (unsafeFromText "TICK") "https://12345"
+        "ed25519_pk15vz9yc5c3upgze8tg5kd7kkzxqgqfxk5a3kudp22hdg0l2za00sq2ufkk7"
+    ]
+    
+absentPid :: PoolId
+absentPid = PoolId "pk192m4ytl5k357e2l666yleuwjlurmf0vxjyh4atxzu5m22q6mexlsp88k7x"
+
+{-------------------------------------------------------------------------------
+                                    Property
+-------------------------------------------------------------------------------}
+
+data TestCase = TestCase
+    { stakePools :: [(PoolId, StakePoolEntry)]
+    , poolIds :: [PoolId]
+    , topDir :: FilePath
+    } deriving (Show, Eq)
+
+data StakePoolEntry
+    = Junk
+    | Meta StakePoolMetadata
+    deriving (Show, Eq)
+
+prop_getStakePoolMetadata :: TestCase -> Property
+prop_getStakePoolMetadata tc = monadicIO $ do
```

Comment:
**[change required]**

Add a comment about what this property is actually checking?

**Why**

the property is rather long, so it's not quite obvious at first glance what is happening. I assume it checks that the logs somehow matches what we expect from parsing files containing both valid and invalid data?